### PR TITLE
post: eslint migration

### PR DESCRIPTION
**Pull request by cexbrayat**

### CODE REVIEW

**Review by jnizet — CHANGES_REQUESTED** [review]
Great post! I added some comments.
I tried to follow the instructions for globe42 (see https://github.com/Ninja-Squad/globe42/pull/1360). I had to add a few exclusions to avoid being overwhelmed, and also had to customize the prettier config otherwise prettier would reject the DOCTYPE declaration in index.html.
I also couldn't remove dependencies that your post suggests to remove (see the comments).
If I understood correctly, eslint (with that configuration) delegates to prettier, which allowed me to remove one of the tasks from the frontend gradle build, since ng lint executes prettier. That's quite nice!

File: _posts/2021-03-30-migrating-from-tslint-to-eslint.md [hunk 1]
```diff
@@ -0,0 +1,175 @@
...
+
+This uses the recommended set of rules from ESlint, @typescript-eslint and @angular-eslint.
+I also added Prettier!
+For this to work, you'll have add `eslint-config-prettier` and `eslint-plugin-prettier`
+as dependencies.
+
+Now when running `ng lint --fix`, ESLint will check all the recommended rules and delegate the formatting to Prettier.
+
+If you opt for this simplified configuration,
+you'll be able to remove a bunch of now useless eslint dependencies (`eslint-plugin-import`, `eslint-plugin-jsdoc`, `eslint-plugin-prefer-arrow`, `@angular-eslint/schematics`).
```
[the remark this answers is not in the document]
Comment (cexbrayat) [reply _posts/2021-03-30-migrating-from-tslint-to-eslint.md:140]: I don't think we have issues in the other projects where I removed `schematics` (it's supposed to be here only to help fr the migration). Yeah, there's an annoying peer deps warning, but that's better than dragging the useless packages :)

**Comment by cexbrayat** [comment]
I forgot something for the DOCTYPE issue, I had the same!

You need to add to prettier:
```
overrides:
  - files: "*.component.html"
    options:
      parser: angular
  - files: "*.html"
    options:
      parser: html
```

So it does not use the Angular parser for all HTML files.
I'll add it in the blog post; Thanks for the review!

File: _posts/2021-03-30-migrating-from-tslint-to-eslint.md (hunk 1 above)
[the remark this answers is not in the document]
Comment (jnizet) [reply _posts/2021-03-30-migrating-from-tslint-to-eslint.md:140]: I missed the eslint-config-prettier dependency as you noted. Once added, everything is fine (except the peer deps) 👍

**Pull request merged**